### PR TITLE
Added Crin keyboard.

### PR DESCRIPTION
**Pull request by KnoblesseOblige**
Added Crin keyboard by Bachoo.

## Description

See above.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crin/crin.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "crin.h"
```
Comment (drashna) [inline keyboards/crin/crin.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/crin/crin.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#pragma once
```
Comment (drashna) [inline keyboards/crin/crin.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/crin/readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Bachoo Crin
+
+Crin is an F-rowless 65%.
+
+* Keyboard Maintainer: [KnoblesseOblige](https://github.com/KnoblesseOblige)
+* Hardware Supported: STM32F303CBT
+* Hardware Availability: TBD
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/crin/readme.md:9]: Could you add instructions on how to enter the bootloader to flash new firmware?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/crin/crin.c (hunk 1 above)
Comment (KnoblesseOblige) [reply keyboards/crin/crin.c:18]: I have added the requested header.

File: keyboards/crin/crin.h (hunk 2 above)
Comment (KnoblesseOblige) [reply keyboards/crin/crin.h:18]: I have added the requested header.

File: keyboards/crin/readme.md (hunk 3 above)
Comment (KnoblesseOblige) [reply keyboards/crin/readme.md:9]: I have added some instructions to the README for entering bootloader. Please let me know if they are too verbose.

**Review by drashna — APPROVED** [review]

File: keyboards/crin/rules.mk [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes   # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = yes    # Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes       # USB Nkey Rollover
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/crin/rules.mk:15]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/crin/rules.mk [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = STM32F303
```
Comment (fauxpark) [inline keyboards/crin/rules.mk:2]:
```suggestion
MCU = STM32F303

# Bootloader selection
BOOTLOADER = stm32-dfu
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**